backend/app/controllers/uoas.js

```diff
@@ -149,22 +156,70 @@ module.exports = {
     deleteOne: function (req, res, next) {
         var thunkQuery = req.thunkQuery;
         co(function* () {
-            var result = yield thunkQuery(UnitOfAnalysisTagLink.select().where(UnitOfAnalysisTagLink.uoaId.equals(req.params.id)));
-            if (_.first(result)) {
-                throw new HttpError(403, 'Subject used in Subject to Tag link. Could not delete Subject');
-            }
 
-            yield thunkQuery(
-                'DELETE FROM "ProductUOA" WHERE "ProductUOA"."productId" = ' +
-                req.body.productId + ' AND "ProductUOA"."UOAid" = ' + req.params.id
-            );
+            if (req.body.productId) {
+                var result = yield thunkQuery(UnitOfAnalysisTagLink.select().where(UnitOfAnalysisTagLink.uoaId.equals(req.params.id)));
+                if (_.first(result)) {
+                    throw new HttpError(403, 'Subject used in Subject to Tag link. Could not delete Subject');
+                }
 
-            yield thunkQuery(
-                'DELETE FROM "Tasks" WHERE "Tasks"."productId" = ' +
-                req.body.productId + ' AND "Tasks"."uoaId" = ' + req.params.id
-            );
+                var task = yield thunkQuery(
+                    'SELECT "Tasks".* ' +
+                    'FROM "Tasks" ' +
+                    'WHERE "Tasks"."uoaId" = ' + req.params.id +
+                    'AND "Tasks"."productId" = ' + req.body.productId +
+                    'AND "Tasks"."isComplete" is True '
+                );
 
-            yield thunkQuery(UnitOfAnalysis.delete().where(UnitOfAnalysis.id.equals(req.params.id)));
+                if (!_.first(task)) {
+
+                    // Soft delete the UOA from the Product UAO Table
+                    yield thunkQuery(
+                        'UPDATE "ProductUOA"' +
+                        'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                        '/ 1000.0)) WHERE "UOAid" = ' + req.params.id +
+                        'AND "productId" = ' + req.body.productId
+                    );
+
+                    // Soft delete the task with that UAO ID
+                    yield thunkQuery(
+                        'UPDATE "Tasks"' +
+                        'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                        '/ 1000.0)) WHERE "productId" = ' + req.body.productId +
+                        'AND "uoaId" = ' + req.params.id
+                    );
+
+                    // Soft delete from the UOA table
+                    yield thunkQuery(
```

Comment:
If we are going to use this endpoint to disassociate uoas from products, this should not set `isDeleted` on the uoa itself when a `productId` is provided, just the ProductUOA.